### ISSUE TITLE
dump schema should also take arrays into account

### ISSUE THREAD

**Issue by cheesycod** [piccolo-orm/piccolo#518]
right now, dump schema automatically doesn't take arrays into account making it painful

**Comment by dantownsend** [comment]
Good point - this needs adding. I think we left it out originally as the array type is a bit more complex than the others.

**Comment by cheesycod** [comment]
yes, still not done

**Comment by cheesycod** [comment]
any updates

**Comment by sinisaos** [comment]
I think this is problematic for generating `Array` columns from an existing (legacy) database because Piccolo uses [_setup_base_column](https://github.com/piccolo-orm/piccolo/blob/46b982be51828da45484d0bfc5e775621660421b/piccolo/table.py#L310-L312) to create the `Array` column at runtime. The easiest way is to manually write the `Array` column (replace the `Column` placeholder) into the generated file and let Piccolo do the setup internally. Not even Django or SQLAlchemy ([sqlacodegen](https://github.com/agronholm/sqlacodegen) is external tool) do this automatically. I'm closing this, but it can be reopened if needed.
